<?xml version="1.0"?>
<robot xmlns:xacro="http://wiki.ros.org/xacro" name= "robot">
  <!-- define pi -->
  <xacro:property name="pi" value="3.14159265359" />
  <!-- links: main serial chain -->
  <link name="base_link">
    <visual>
      <origin xyz="0 0 0.15" rpy="0 0 0"/>
      <geometry>
        <box size="0.17 0.17 0.3"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0.15" rpy="0 0 0"/>
      <geometry>
        <box size="0.17 0.17 0.3"/>
      </geometry>
    </collision>
  </link>
  <link name="base_link_inertia">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 3.14159265359" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/base.stl" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 3.14159265359" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/base.stl" />
      </geometry>
    </collision>
  </link>
  <link name="shoulder_link">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 3.14159265359" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/shoulder.stl" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 3.14159265359" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/shoulder.stl" />
      </geometry>
    </collision>
  </link>
  <link name="upper_arm_link">
    <visual>
      <origin xyz="0 0  0.220941" rpy="1.5707963267948966 0 -1.5707963267948966" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/upperarm.stl" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0.220941" rpy="1.5707963267948966 0 -1.5707963267948966" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/upperarm.stl" />
      </geometry>
    </collision>
  </link>
  <link name="forearm_link">
    <visual>
      <origin xyz="0 0 0.049042" rpy="1.5707963267948966 0 -1.5707963267948966" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/forearm.stl" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0  0.049042" rpy="1.5707963267948966 0 -1.5707963267948966" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/forearm.stl" />
      </geometry>
    </collision>
  </link>
  <link name="wrist_1_link">
    <visual>
      <origin xyz="0 0 -0.1149" rpy="1.5707963267948966 0 0" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/wrist1.stl" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 -0.1149" rpy="1.5707963267948966 0 0" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/wrist1.stl" />
      </geometry>
    </collision>
  </link>
  <link name="wrist_2_link">
    <visual>
      <origin xyz="0 0 -0.1158" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/wrist2.stl" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 -0.1158" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/wrist2.stl" />
      </geometry>
    </collision>
  </link>
  <link name="wrist_3_link">
    <visual>
      <origin xyz="0 0 -0.0922" rpy="1.5707963267948966 0 0" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/wrist3.stl" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 -0.0922" rpy="1.5707963267948966 0 0" />
      <geometry>
        <mesh filename="package://manipulacion_pkg/urdf/meshes/collision/wrist3.stl" />
      </geometry>
    </collision>
  </link>

  <!-- joints: main serial chain -->
  <joint name="base_link-base_link_inertia" type="fixed">
    <parent link="base_link" />
    <child link="base_link_inertia" />
    <origin xyz="0 0 0.32" rpy="0 0 3.14159265359" />
  </joint>
  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link_inertia" />
    <child link="shoulder_link" />
    <origin xyz="0 0 0.1273" rpy="0 0 0" />
    <axis xyz="0 0 1" />
    <limit lower="-3.1415" upper="3.1415"
      effort="330" velocity="2.0943951023931953" />
    <dynamics damping="0" friction="0" />
  </joint>
  <joint name="shoulder_lift_joint" type="revolute">
    <parent link="shoulder_link" />
    <child link="upper_arm_link" />
    <origin xyz="0 0 0" rpy="1.570796327 0 0" />
    <axis xyz="0 0 1" />
    <limit lower="-6.2831830605385" upper="6.2831830605385"
      effort="330" velocity="2.0943951023931953" />
    <dynamics damping="0" friction="0" />
  </joint>
  <joint name="elbow_joint" type="revolute">
    <parent link="upper_arm_link" />
    <child link="forearm_link" />
    <origin xyz="-0.612 0 0" rpy="0 0 0" />
    <axis xyz="0 0 1" />
    <limit lower="-3.141592653589793" upper="3.141592653589793"
      effort="150" velocity="3.141592653589793" />
    <dynamics damping="0" friction="0" />
  </joint>
  <joint name="wrist_1_joint" type="revolute">
    <parent link="forearm_link" />
    <child link="wrist_1_link" />
    <origin xyz="-0.5723 0 0.163941" rpy="0 0 0" />
    <axis xyz="0 0 1" />
    <limit lower="-6.2831830605385" upper="6.2831830605385"
      effort="56" velocity="3.141592653589793" />
    <dynamics damping="0" friction="0" />
  </joint>
  <joint name="wrist_2_joint" type="revolute">
    <parent link="wrist_1_link" />
    <child link="wrist_2_link" />
    <origin xyz="0 -0.1157 -2.373046667922381e-11" rpy="1.570796327 0 0" />
    <axis xyz="0 0 1" />
    <limit lower="-6.2831830605385" upper="6.2831830605385"
      effort="56" velocity="3.141592653589793" />
    <dynamics damping="0" friction="0" />
  </joint>
  <joint name="wrist_3_joint" type="revolute">
    <parent link="wrist_2_link" />
    <child link="wrist_3_link" />
    <origin xyz="0 0.0922 -1.891053610911353e-11"
      rpy="1.570796326589793 3.141592653589793 3.14159265358979" />
    <axis xyz="0 0 1" />
    <limit lower="-6.2831830605385" upper="6.2831830605385"
      effort="56" velocity="3.141592653589793" />
    <dynamics damping="0" friction="0" />
  </joint>

  <!-- ROS-Industrial 'base' frame: base_link to UR 'Base' Coordinates transform -->
  <!--link
  name="${prefix}base"/>
    <joint name="${prefix}base_link-base_fixed_joint" type="fixed">

      <origin xyz="0 0 0" rpy="0 0 ${pi}"/>
      <parent link="${prefix}base_link"/>
      <child link="${prefix}base"/>
    </joint-->

  <link name="flange" />
  <joint name="wrist_3-flange" type="fixed">
    <parent link="wrist_3_link" />
    <child link="flange" />
    <origin xyz="0 0 0" rpy="0 -1.5707963267948966 -1.5707963267948966" />
  </joint>

  <link name="tool0" />
  <joint name="flange-tool0" type="fixed">
    <origin xyz="0 0 0" rpy="1.5707963267948966 0 1.5707963267948966" />
    <parent link="flange" />
    <child link="tool0" />
  </joint>
  <link name="hand"/>
  <joint name="tool0-hand" type="fixed">
    <origin xyz="0 0 0" rpy="0 0 0" />
    <parent link="tool0" />
    <child link="hand" />
  </joint>


</robot>